assets/src/edit-story/components/dropTargets/provider.js

```diff
@@ -174,26 +175,49 @@ function DropTargetsProvider({ children }) {
           resource,
         });
       }
-      combineElements(combineArgs);
-
-      // Reset styles on visisble elements
-      elements
-        .filter(({ id }) => !(id in Object.keys(dropTargets)) && id !== selfId)
-        .forEach((el) => {
-          pushTransform(el.id, {
-            dropTargets: {
-              active: false,
-              replacement: null,
+      const firstElement = elements.find(
+        ({ id }) => id === combineArgs.firstId
+      );
+      const finalizeDrop = (baseColor) => {
+        if (baseColor && !combineArgs.firstElement) {
+          combineArgs.firstElement = {
+            ...firstElement,
+            resource: {
+              ...firstElement.resource,
+              baseColor,
             },
+          };
+          delete combineArgs.firstId;
+        }
+        combineElements(combineArgs);
+
+        // Reset styles on visible elements
+        elements
+          .filter(
+            ({ id }) => !(id in Object.keys(dropTargets)) && id !== selfId
+          )
+          .forEach((el) => {
+            pushTransform(el.id, {
+              dropTargets: {
+                active: false,
+                replacement: null,
+              },
+            });
+            pushTransform(el.id, null);
           });
-          pushTransform(el.id, null);
-        });
 
-      setActiveDropTargetId(null);
+        setActiveDropTargetId(null);
 
-      const { onDropHandler } = getDefinitionForType(resource.type);
-      if (onDropHandler) {
-        onDropHandler(activeDropTargetId);
+        const { onDropHandler } = getDefinitionForType(resource.type);
+        if (onDropHandler) {
+          onDropHandler(activeDropTargetId);
+        }
+      };
+      // Skip if we already have the color.
+      if (firstElement.baseColor) {
```

Comment:
This by the way fails (`Cannot read property 'baseColor' of undefined`), when you drop an element directly from the library to the background drop target.